src/niscope/templates/session.py/fancy_fetch.py.mako

```diff
@@ -21,7 +21,10 @@
 
         wfm, wfm_info = self._${f['python_name']}(num_samples, timeout)
 
-        mv = memoryview(wfm)
+        if isinstance(wfm, array.ArrayType):
```

Comment:
This update was to keep the `memoryview` instantiation for the `wfm` when it was coming from the `_library_interpreter`. But when coming from grpc, we should be able to just pass the returned `wfm` forward to the `_populate_samples_info` method.

Reply:
I've double-checked the C API documentation and thought through this and it makes sense to me. And I trust that our tests would catch if we weren't handling the data right.